### PR TITLE
Refactor mailer ally request and accepted request

### PR DESCRIPTION
**Pull request by chrisrp**
* Extracts ally notifications
* Adds I18n to notifications
* Adds UrlHelper module

The idea is to decompose the notification_mailer to smaller pieces. I am not sure if it is really necessary to group by types of notification with namespaces like `GroupType::EventType` but it is easy to change later if it does not make sense. 

Eg. `AllyNotifications::AcceptedAllyRequest` and `AllyNotifications::NewAllyRequest`.

### CODE REVIEW

File: app/mailers/ally_notifications/new_ally_request.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    def to
+      @recipient.email
+    end
+
+    def subject
+      I18n.t('mailers.new_ally_request.subject', ally_name: @data[:user])
+    end
+
+    def message
+      I18n.t('mailers.new_ally_request.message', allies_url: allies_url)
```
Comment (julianguyen) [inline app/mailers/ally_notifications/new_ally_request.rb]: How is `allies_url` being passed here?

File: app/mailers/notification_mailer.rb [hunk 2]
```diff
@@ -18,7 +18,7 @@ def refill_medication(reminder)
   end
 
   def notification_email(recipientid, data)
-    @data = JSON.parse(data)
+    @data = HashWithIndifferentAccess.new(JSON.parse(data))
     @recipient = User.where(id: recipientid).first
     subject = 'if me | '
```
Comment (julianguyen) [inline app/mailers/notification_mailer.rb]: Should be moved into i18n

File: config/application.rb [hunk 3]
```diff
@@ -24,7 +24,7 @@ class Application < Rails::Application
 
     config.exceptions_app = self.routes
 
-    config.autoload_paths += Dir[Rails.root.join('app', 'uploaders')]
+    config.autoload_paths << Rails.root.join('lib')
```
Comment (julianguyen) [inline config/application.rb]: Could you explain what this does? Not quite sure (:

**Comment by julianguyen** [comment]
Awesome refactor @chrisrp! 💯 
Left a few comments, otherwise LGTM. Ooh also try to address these https://codeclimate.com/github/julianguyen/ifme/pull/309 the best you can.

File: app/mailers/ally_notifications/new_ally_request.rb (hunk 1 above)
Comment (chrisrp) [inline app/mailers/ally_notifications/new_ally_request.rb]: Because it includes the ` ::UrlHelper`

File: app/mailers/notification_mailer.rb (hunk 2 above)
Comment (chrisrp) [inline app/mailers/notification_mailer.rb]: Sure, but the way I thought once all the notifications get extracted they will have the full subject.

Comment (julianguyen) [inline app/mailers/notification_mailer.rb]: Sounds good. We can do this for now then, and then in another PR do that.

**Pull request merged**